tests/unit/build_providers/multipass/test_multipass.py

```diff
@@ -747,6 +746,36 @@ def test_lifecycle(
                 hide_output=False,
                 instance_name="snapcraft-project-name",
             ),
+            call(
```

Comment:
We wouldn't need this with a shim like https://github.com/snapcore/snapcraft/pull/3251/files#diff-658f276fdea7891f007be4110b902a42R101 applied to https://github.com/snapcore/snapcraft/pull/3251/files#diff-658f276fdea7891f007be4110b902a42R106 to only the multipass specific bits of the implementation

Reply:
Well this whole test case appears to be about testing the whole thing... the LXD version uses `asserts_has_calls()`, but also enumerates the whole thing (minus these new changes because it never failed).  If the purpose of the test is to simply check that it has any calls, and that each call has the proper provider-specific prefix commands, I can do that.  Otherwise, I'm not sure?

Reply:
I was trying to transform it into that for changes not to have this ripple effect. It is true that it is not the case for everything today

Reply:
Pushed updates to clean that up in areas touched :+1: